### PR TITLE
Match specific tolerations to prevent Diffs

### LINKED ISSUE #955: My toleration is not saved to the tfstate at the end of both successful or failed terraform apply
### Terraform Version and Provider Version

Terraform v0.13.0
+ provider registry.terraform.io/-/kubernetes v1.12.0
+ provider registry.terraform.io/hashicorp/aws v3.1.0
+ provider registry.terraform.io/hashicorp/kubernetes v1.12.0

### Affected Resource(s)

- kubernetes_deployment

### Terraform Configuration Files
```hcl

provider "aws" {
  region = "eu-west-1"
}

provider "kubernetes" {
  config_path = HIDDEN
}

terraform {
  backend "s3" {
    bucket         = "HIDDEN"
    key            = "HIDDEN/terraform.tfstate"
    region         = "eu-west-1"
    encrypt        = "true"
    dynamodb_table = "HIDDEN"
  }
}

resource "kubernetes_service_account" "test_nginx" {
  metadata {
    name      = "test-nginx"
    namespace = "kube-system"
  }
  automount_service_account_token = true

}

resource "kubernetes_deployment" "test_nginx" {
  depends_on = [kubernetes_service_account.test_nginx]

  metadata {
    name      = "test-nginx"
    namespace = "kube-system"
    labels = {
      app = "nginx"
    }
  }

  spec {
    replicas = 2
    selector {
      match_labels = {
        app = "nginx"
      }
    }
    template {
      metadata {
        labels = {
          app = "nginx"
        }
      }

      spec {
        automount_service_account_token  = true
        dns_policy                       = "Default"
        host_ipc                         = false
        host_network                     = false
        host_pid                         = false
        node_selector                    = {}
        priority_class_name              = "system-cluster-critical"
        restart_policy                   = "Always"
        service_account_name             = "test-nginx"
        share_process_namespace          = false
        termination_grace_period_seconds = 30

        container {
          image = "nginx:1.7.9"
          name  = "nginx"

          port {
            container_port = 80
          }
        }

        toleration {
          effect   = "NoSchedule"
          key      = "node.kubernetes.io/role"
          operator = "Equal"
          value    = "admin"
        }
      }

    }
  }
}


```

### Debug Output

```

$ terraform apply -auto-approve
[id=75cf6a0832847961abe3a6c5905fe886ced5a3fa6c8f5b0a50c7577d3ea3923f]
kubernetes_service_account.test_nginx: Creating...
kubernetes_service_account.test_nginx: Creation complete after 0s [id=kube-system/test-nginx]
kubernetes_deployment.test_nginx: Creating...
kubernetes_deployment.test_nginx: Creation complete after 5s [id=kube-system/test-nginx]

Apply complete! Resources: 2 added, 0 changed, 0 destroyed.
Releasing state lock. This may take a few moments...

```
### Expected Behavior

We should be seing a deployment called test-nginx in kube-system with the toleration:

```
$ kubectl get deployments test-nginx -n kube-system -o yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    deployment.kubernetes.io/revision: "1"
  creationTimestamp: 2020-08-12T13:04:27Z
  generation: 1
  labels:
    app: nginx
  name: test-nginx
  namespace: kube-system
  resourceVersion: "4647985"
  selfLink: /apis/apps/v1/namespaces/kube-system/deployments/test-nginx
  uid: ed60438e-020b-47cb-9e9a-c2f2f0355427
spec:
  progressDeadlineSeconds: 600
  replicas: 2
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      app: nginx
  strategy:
    rollingUpdate:
      maxSurge: 25%
      maxUnavailable: 25%
    type: RollingUpdate
  template:
    metadata:
      creationTimestamp: null
      labels:
        app: nginx
    spec:
      automountServiceAccountToken: true
      containers:
      - image: nginx:1.7.9
        imagePullPolicy: IfNotPresent
        name: nginx
        ports:
        - containerPort: 80
          protocol: TCP
        resources: {}
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
      dnsPolicy: Default
      priorityClassName: system-cluster-critical
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      serviceAccount: test-nginx
      serviceAccountName: test-nginx
      shareProcessNamespace: false
      terminationGracePeriodSeconds: 30
      tolerations:
      - effect: NoSchedule
        key: node.kubernetes.io/role
        operator: Equal
        value: admin
status:
  availableReplicas: 2
  conditions:
  [...]
  readyReplicas: 2
  replicas: 2
  updatedReplicas: 2

```
And the resource in terraform tfstate with the toleration too:

```
$terraform` state show kubernetes_deployment.test_nginx
# kubernetes_deployment.test_nginx:
resource "kubernetes_deployment" "test_nginx" {
    id               = "kube-system/test-nginx"
    wait_for_rollout = true

    metadata {
        generation       = 1
        labels           = {
            "app" = "nginx"
        }
        name             = "test-nginx"
        namespace        = "kube-system"
        resource_version = "4647985"
        self_link        = "/apis/apps/v1/namespaces/kube-system/deployments/test-nginx"
        uid              = "ed60438e-020b-47cb-9e9a-c2f2f0355427"
    }

    spec {
        min_ready_seconds         = 0
        paused                    = false
        progress_deadline_seconds = 600
        replicas                  = 2
        revision_history_limit    = 10

        selector {
            match_labels = {
                "app" = "nginx"
            }
        }

        strategy {
            type = "RollingUpdate"

            rolling_update {
                max_surge       = "25%"
                max_unavailable = "25%"
            }
        }

        template {
            metadata {
                generation = 0
                labels     = {
                    "app" = "nginx"
                }
            }

            spec {
                active_deadline_seconds          = 0
                automount_service_account_token  = true
                dns_policy                       = "Default"
                host_ipc                         = false
                host_network                     = false
                host_pid                         = false
                priority_class_name              = "system-cluster-critical"
                restart_policy                   = "Always"
                service_account_name             = "test-nginx"
                share_process_namespace          = false
                termination_grace_period_seconds = 30

                container {
                    image                    = "nginx:1.7.9"
                    image_pull_policy        = "IfNotPresent"
                    name                     = "nginx"
                    stdin                    = false
                    stdin_once               = false
                    termination_message_path = "/dev/termination-log"
                    tty                      = false

                    port {
                        container_port = 80
                        host_port      = 0
                        protocol       = "TCP"
                    }

                    resources {
                    }
                }

                toleration {
                    effect   = "NoSchedule"
                    key      = "node.kubernetes.io/role"
                    operator = "Equal"
                    value    = "admin"
                }
            }
        }
    }
}

```

### Actual Behavior

Resource deployment is created with the toleration, and both pods are deployed in nodes with label "admin":

```
$ kubectl get deployments test-nginx -n kube-system -o yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    deployment.kubernetes.io/revision: "1"
  creationTimestamp: 2020-08-12T13:04:27Z
  generation: 1
  labels:
    app: nginx
  name: test-nginx
  namespace: kube-system
  resourceVersion: "4647985"
  selfLink: /apis/apps/v1/namespaces/kube-system/deployments/test-nginx
  uid: ed60438e-020b-47cb-9e9a-c2f2f0355427
spec:
  progressDeadlineSeconds: 600
  replicas: 2
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      app: nginx
  strategy:
    rollingUpdate:
      maxSurge: 25%
      maxUnavailable: 25%
    type: RollingUpdate
  template:
    metadata:
      creationTimestamp: null
      labels:
        app: nginx
    spec:
      automountServiceAccountToken: true
      containers:
      - image: nginx:1.7.9
        imagePullPolicy: IfNotPresent
        name: nginx
        ports:
        - containerPort: 80
          protocol: TCP
        resources: {}
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
      dnsPolicy: Default
      priorityClassName: system-cluster-critical
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      serviceAccount: test-nginx
      serviceAccountName: test-nginx
      shareProcessNamespace: false
      terminationGracePeriodSeconds: 30
      tolerations:
      - effect: NoSchedule
        key: node.kubernetes.io/role
        operator: Equal
        value: admin
status:
  availableReplicas: 2
  conditions:
  - lastTransitionTime: 2020-08-12T13:04:29Z
    lastUpdateTime: 2020-08-12T13:04:29Z
    message: Deployment has minimum availability.
    reason: MinimumReplicasAvailable
    status: "True"
    type: Available
  - lastTransitionTime: 2020-08-12T13:04:27Z
    lastUpdateTime: 2020-08-12T13:04:29Z
    message: ReplicaSet "test-nginx-c4ddd86" has successfully progressed.
    reason: NewReplicaSetAvailable
    status: "True"
    type: Progressing
  observedGeneration: 1
  readyReplicas: 2
  replicas: 2
  updatedReplicas: 2

$ kubectl get deployments test-nginx -n kube-system
NAME         READY   UP-TO-DATE   AVAILABLE   AGE
test-nginx   2/2     2            2           10m

```
But the toleration is missing in the tfstate: 

```
$ terraform state show kubernetes_deployment.test_nginx
# kubernetes_deployment.test_nginx:
resource "kubernetes_deployment" "test_nginx" {
    id               = "kube-system/test-nginx"
    wait_for_rollout = true

    metadata {
        generation       = 1
        labels           = {
            "app" = "nginx"
        }
        name             = "test-nginx"
        namespace        = "kube-system"
        resource_version = "4647985"
        self_link        = "/apis/apps/v1/namespaces/kube-system/deployments/test-nginx"
        uid              = "ed60438e-020b-47cb-9e9a-c2f2f0355427"
    }

    spec {
        min_ready_seconds         = 0
        paused                    = false
        progress_deadline_seconds = 600
        replicas                  = 2
        revision_history_limit    = 10

        selector {
            match_labels = {
                "app" = "nginx"
            }
        }

        strategy {
            type = "RollingUpdate"

            rolling_update {
                max_surge       = "25%"
                max_unavailable = "25%"
            }
        }

        template {
            metadata {
                generation = 0
                labels     = {
                    "app" = "nginx"
                }
            }

            spec {
                active_deadline_seconds          = 0
                automount_service_account_token  = true
                dns_policy                       = "Default"
                host_ipc                         = false
                host_network                     = false
                host_pid                         = false
                priority_class_name              = "system-cluster-critical"
                restart_policy                   = "Always"
                service_account_name             = "test-nginx"
                share_process_namespace          = false
                termination_grace_period_seconds = 30

                container {
                    image                    = "nginx:1.7.9"
                    image_pull_policy        = "IfNotPresent"
                    name                     = "nginx"
                    stdin                    = false
                    stdin_once               = false
                    termination_message_path = "/dev/termination-log"
                    tty                      = false

                    port {
                        container_port = 80
                        host_port      = 0
                        protocol       = "TCP"
                    }

                    resources {
                    }
                }
            }
        }
    }
}

```
And so, when I execute again "terraform plan" with the same code, terraform tries to update the deployment with the toleration again (as you can see above, the output of terraform apply was, in fact, successful):
 
```
$ terraform plan 
Acquiring state lock. This may take a few moments...
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

[id=75cf6a0832847961abe3a6c5905fe886ced5a3fa6c8f5b0a50c7577d3ea3923f]
kubernetes_service_account.test_nginx: Refreshing state... [id=kube-system/test-nginx]
kubernetes_deployment.test_nginx: Refreshing state... [id=kube-system/test-nginx]

------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # kubernetes_deployment.test_nginx will be updated in-place
  ~ resource "kubernetes_deployment" "test_nginx" {
        id               = "kube-system/test-nginx"
        wait_for_rollout = true

        metadata {
            annotations      = {}
            generation       = 1
            labels           = {
                "app" = "nginx"
            }
            name             = "test-nginx"
            namespace        = "kube-system"
            resource_version = "4647985"
            self_link        = "/apis/apps/v1/namespaces/kube-system/deployments/test-nginx"
            uid              = "ed60438e-020b-47cb-9e9a-c2f2f0355427"
        }

      ~ spec {
            min_ready_seconds         = 0
            paused                    = false
            progress_deadline_seconds = 600
            replicas                  = 2
            revision_history_limit    = 10

            selector {
                match_labels = {
                    "app" = "nginx"
                }
            }

            strategy {
                type = "RollingUpdate"

                rolling_update {
                    max_surge       = "25%"
                    max_unavailable = "25%"
                }
            }

          ~ template {
                metadata {
                    annotations = {}
                    generation  = 0
                    labels      = {
                        "app" = "nginx"
                    }
                }

              ~ spec {
                    active_deadline_seconds          = 0
                    automount_service_account_token  = true
                    dns_policy                       = "Default"
                    host_ipc                         = false
                    host_network                     = false
                    host_pid                         = false
                    node_selector                    = {}
                    priority_class_name              = "system-cluster-critical"
                    restart_policy                   = "Always"
                    service_account_name             = "test-nginx"
                    share_process_namespace          = false
                    termination_grace_period_seconds = 30

                    container {
                        args                     = []
                        command                  = []
                        image                    = "nginx:1.7.9"
                        image_pull_policy        = "IfNotPresent"
                        name                     = "nginx"
                        stdin                    = false
                        stdin_once               = false
                        termination_message_path = "/dev/termination-log"
                        tty                      = false

                        port {
                            container_port = 80
                            host_port      = 0
                            protocol       = "TCP"
                        }

                        resources {
                        }
                    }

                  + toleration {
                      + effect   = "NoSchedule"
                      + key      = "node.kubernetes.io/role"
                      + operator = "Equal"
                      + value    = "admin"
                    }
                }
            }
        }
    }

Plan: 0 to add, 1 to change, 0 to destroy.

```

I have also tried to update the tfsate with the toleration (terraform pull, edit toleration and serial, terraform push), but I get same results:  when I execute "terraform plan" the toleration must be added and besides, when I do "terraform apply", my manually-added toleration is removed from the tfstate.

### Steps to Reproduce
1. `terraform apply` (ok)  
2. `terraform state show kubernetes_deployment.test_nginx` (toleration is missing)
3. `terraform plan` (toleration to be added)
4. Back to point 1 (even if toleration is already added in the test-nginx deployment).

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### PR DESCRIPTION
**Pull request by DrFaust92**
### Acceptance tests
- [x] Have you added an acceptance test for the functionality being added?
- [x] Have you run the acceptance tests on this branch? (If so, please include the test log in a gist)

```
--- PASS: TestAccKubernetesDeployment_with_tolerations (54.41s)
--- PASS: TestAccKubernetesDeployment_with_tolerations_unset_toleration_seconds (11.52s)

--- PASS: TestAccKubernetesDaemonSet_with_tolerations (5.58s)
--- PASS: TestAccKubernetesDaemonSet_with_tolerations_unset_toleration_seconds (6.17s)
```

### References

Closes #955

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### CODE REVIEW

**Comment by DrFaust92** [comment]
rebased

**Review by dak1n1 — COMMENTED** [review]
This looks good! I did some manual testing and ran the full set of acceptance tests on it, and everything passed. I just left one suggestion.

File: kubernetes/structures_pod.go [hunk 1]
```diff
@@ -4,13 +4,23 @@ import (
...
 
+// https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/#taint-based-evictions
+var builtInTolerations = map[string]string{
+	"node.kubernetes.io/not-ready":           "",
+	"node.kubernetes.io/unreachable":         "",
+	"node.kubernetes.io/out-of-disk":         "",
+	"node.kubernetes.io/memory-pressure":     "",
+	"node.kubernetes.io/disk-pressure":       "",
+	"node.kubernetes.io/network-unavailable": "",
+	"node.kubernetes.io/unschedulable":       "",
```
[the remark this answers is not in the document]
Comment (DrFaust92) [reply kubernetes/structures_pod.go:21]: Nice, will change

[the remark this answers is not in the document]
Comment (DrFaust92) [reply kubernetes/structures_pod.go]: done.

**Review by dak1n1 — APPROVED** [review]
Thanks for this contribution!

**Pull request merged**